pxr/usd/sdf/layerRegistry.h

```diff
@@ -91,74 +88,58 @@ class Sdf_LayerRegistry
     SdfLayerHandle _FindByRealPath(
         const std::string& layerPath,
         const std::string& resolvedPath=std::string()) const;
-    
-    // Index tags.
-    struct by_identity {};
-    struct by_identifier {};
-    struct by_repository_path {};
-    struct by_real_path {};
-
-    // Key Extractors.
-    struct layer_identifier {
-        typedef std::string result_type;
-        const result_type& operator()(const SdfLayerHandle& layer) const;
-    };
 
-    struct layer_repository_path {
-        typedef std::string result_type;
-        result_type operator()(const SdfLayerHandle& layer) const;
+    // A wrapper around a set of unordered_maps that maps layers
+    // bidirectionally to their various string representations (realPath,
+    // identifier, and repositoryPath)
+    class _Layers final {
+    public:
+        _Layers() = default;
+        using LayersByRealPath =
+            std::unordered_map<std::string, SdfLayerHandle, TfHash>;
```

Comment:
I don't believe that LayersByIdentifier and LayersByRepositoryPath are going to be equivalent to the original behavior if implemented as std::unordered_map as the multi index container had those two keys set as non-unique.
I'm trying to find out from internal conversations as to if/why this is necessary.

Reply:
This is a good catch!  I think "one => many" can be similarly modeled with a `unordered_multimap`, but it'd would be good to understand the practical situations where the `unordered_map` implementation fails so we can get test coverage for it.

Reply:
I've learned at least one of the reasons is that you can have different identifiers resolve to the same real layer path under different ArResolverContexts. And the good news is we do have a test case for this in testSdfLayer so we should be able to catch if there's an issue pretty easily.

Reply:
`testSdfLayer` passes. It seems like `SdfLayer` tries to avoid looking up identifiers that are context dependent (deferring to the real path instead) which probably explains why the code is working.  It seems like the case that might fail is that if you mixed context dependent paths and non-context dependent paths and the context-dependent path boots the non-context dependent path lookup from the registry.  I'll see if I can fashion that test case and move things over to use `unordered_multimap`.

Reply:
`testSdfLayer` was passing because the test which exercises multiple contexts was being skipped due to 

```
    @unittest.skipIf(preferredResolver != "ArDefaultResolver",
                     "Test uses search-path functionality specific to "
                     "ArDefaultResolver")
```

By default, `testSdfLayer` is configured to run only against a test resolver.

```
preferredResolver = os.environ.get(
    "TEST_SDF_LAYER_RESOLVER", "Sdf_TestResolver")
```